### ISSUE TITLE
r/aws_codepipeline should support action namespace

### ISSUE THREAD

**Issue by munkyboy** [terraform-providers/terraform-provider-aws#11418]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Pipeline supports passing variables between actions. It requires that each action declare a namespace 

### New or Affected Resource(s)

* aws_codepipeline

### Potential Terraform Configuration

```hcl
resource "aws_codepipeline" "main" {
  stage {
    action {
      namespace = "<MyNamespace>"
    }
  }
}
```

### References

* https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-variables.html

**Comment by michel-lacle** [comment]
@munkyboy , I need this feature for my project, so I'd like to take a stab at implementing this feature. I have never contributed to this project, can you point me in the right direction to start?

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 2.59.0 of the Terraform AWS Provider, later this week. Thanks to @suchpuppet for the implementation. 👍
